src/pages/signin/SignInPage.js

```diff
@@ -64,23 +79,49 @@ const defaultProps = {
 /**
  * @param {Boolean} hasLogin
  * @param {Boolean} hasValidateCode
+ * @param {Object} account
  * @param {Boolean} isPrimaryLogin
- * @param {Boolean} isAccountValidated
+ * @param {Boolean} isUsingMagicCode
  * @param {Boolean} hasEmailDeliveryFailure
  * @returns {Object}
  */
-function getRenderOptions({hasLogin, hasValidateCode, hasAccount, isPrimaryLogin, isAccountValidated, hasEmailDeliveryFailure, isClientTheLeader}) {
-    const shouldShowLoginForm = isClientTheLeader && !hasLogin && !hasValidateCode;
-    const shouldShowEmailDeliveryFailurePage = hasLogin && hasEmailDeliveryFailure;
-    const isUnvalidatedSecondaryLogin = hasLogin && !isPrimaryLogin && !isAccountValidated && !hasEmailDeliveryFailure;
-    const shouldShowValidateCodeForm = hasAccount && (hasLogin || hasValidateCode) && !isUnvalidatedSecondaryLogin && !hasEmailDeliveryFailure;
-    const shouldShowWelcomeHeader = shouldShowLoginForm || shouldShowValidateCodeForm || isUnvalidatedSecondaryLogin;
-    const shouldShowWelcomeText = shouldShowLoginForm || shouldShowValidateCodeForm || !isClientTheLeader;
+function getRenderOptions({hasLogin, hasValidateCode, account, isPrimaryLogin, isUsingMagicCode, isClientTheLeader}) {
+    const hasAccount = !_.isEmpty(account);
+    const isSAMLEnabled = Boolean(account.isSAMLEnabled);
+    const isSAMLRequired = Boolean(account.isSAMLRequired);
+    const hasEmailDeliveryFailure = Boolean(account.hasEmailDeliveryFailure);
+
+    // SAML is temporarily restricted to users on the beta or to users signing in on web and mweb
+    let shouldShowChooseSSOOrMagicCode = false;
+    let shouldInitiateSAMLLogin = false;
+    const platform = getPlatform();
+    if (Permissions.canUseSAML() || platform === CONST.PLATFORM.WEB || platform === CONST.PLATFORM.DESKTOP) {
```

Comment:
I didn't notice in tests because is Beta's are on by default.
Should this condition be?
```suggestion
    if (Permissions.canUseSAML() && (platform === CONST.PLATFORM.WEB || platform === CONST.PLATFORM.DESKTOP)) {
```

Reply:
I see beta is for other platforms, thanks for clarifying!